test/food-test.js

```diff
@@ -0,0 +1,232 @@
+          body: food,
+          json: true,
+          url: '/api/v1/foods'
+        }
+
+        const ourRequest = this.request(options, function(error, response, body) {
+          if (error) { done(error) }
+
+          assert.equal(body.name, 'apple')
+          assert.equal(body.calories, 20)
+          done()
+        })
+      })
+    })
+
+    describe('PUT /api/v1/foods/:id', function() {
+      this.timeout(1000000)
+
+      beforeEach( function(done) {
+        Food.create('pizza', 100)
+          .then( function(){ done() })
+        })
+
+      afterEach( function(done) {
+        Food.destroyAll()
+        .then( function() { done() })
+      })
+
+      it('updates the foods name and calories for a food', function(done) {
+        const id = 1
+        const food = { name: 'banana', calories: 10 }
+        const options = {
+          method: 'PUT',
+          body: food,
+          json: true,
+          url: `/api/v1/foods/${id}`
+        }
+        const ourRequest = this.request(options, function(error, response, body) {
+          if (error) { done(error) }
+          assert.equal(body.name, 'banana')
+          assert.equal(body.calories, 10)
+          done()
+        })
+      })
+      it('updates the foods name', function(done) {
+        const id = 1
+        const food = { name: 'banana' }
+        const options = {
+          method: 'PUT',
+          body: food,
+          json: true,
+          url: `/api/v1/foods/${id}`
+        }
+        const ourRequest = this.request(options, function(error, response, body) {
+          if (error) { done(error) }
+          assert.equal(body.name, 'banana')
+          done()
+        })
+      })
+
+      it('updates the calories count', function(done) {
+        const id = 1
+        const food = { calories: 15 }
+        const options = {
+          method: 'PUT',
+          body: food,
+          json: true,
+          url: `/api/v1/foods/${id}`
+        }
+        const ourRequest = this.request(options, function(error, response, body) {
+          if (error) { done(error) }
+          assert.equal(body.calories, 15)
+          done()
+        })
+      })
+    })
+
+    describe('DELETE /api/v1/foods/:id', function() {
+      this.timeout(1000000)
+
+      beforeEach( function(done) {
+        Food.create('pizza', 100)
+          .then( function(){ done() })
+        })
+
+      afterEach( function(done) {
+        Food.destroyAll()
+        .then( function() { done() })
+      })
+
+      it('deletes the food from the database', function(done) {
+        const id = 1
+        const options = {
+          method: 'DELETE',
+          json: true,
+          url: `/api/v1/foods/${id}`
+        }
+        const ourRequest = this.request(options, function(error, response) {
+            if (error) { done(error) }
+            Food.all()
+            .then( (data) => {
+            assert.equal(response.statusCode, 200)
+            assert.equal(data.rowCount, 2)
```

Comment:
Getting a foreign key constraint on the delete action with the new joins table we created so i'm going to have to revisit this tomorrow